src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getMaxId, getUserById, getValidTitle } from './utils/utils';
+
+const initialTodos: Todo[] = todosFromServer.map((todo) => {
+  const user = usersFromServer.find(({ id }) => id === todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title && !userId) {
+      setHasUserIdError(true);
+      setHasTitleError(true);
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+      return;
+    }
+
+    if (!title) {
+      setHasTitleError(true);
+
+      return;
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getMaxId(todos) + 1,
+      title: getValidTitle(title),
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setTodos((prevTodo: Todo[]) => ([...prevTodo, newTodo]));
+    setUserId(0);
+    setTitle('');
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
```

Comment:
Isn't it better to move the form to the separate component, to reduce the complexity of App.tsx?

Reply:
@rostyslav48 is right - please add separate component for the form (inputs + button)